lib/expression.ts

```diff
@@ -11,31 +17,71 @@ export default class ExpressionTranspiler extends base.TranspilerBase {
       case ts.SyntaxKind.BinaryExpression:
         var binExpr = <ts.BinaryExpression>node;
         var operatorKind = binExpr.operatorToken.kind;
-        if (operatorKind === ts.SyntaxKind.EqualsEqualsEqualsToken ||
-            operatorKind === ts.SyntaxKind.ExclamationEqualsEqualsToken) {
-          if (operatorKind === ts.SyntaxKind.ExclamationEqualsEqualsToken) this.emit('!');
-          this.emit('identical (');
-          this.visit(binExpr.left);
-          this.emit(',');
-          this.visit(binExpr.right);
-          this.emit(')');
-        } else {
-          this.visit(binExpr.left);
-          if (operatorKind === ts.SyntaxKind.InstanceOfKeyword) {
+        var tokenStr = ts.tokenToString(operatorKind);
+        switch (operatorKind) {
+          case ts.SyntaxKind.EqualsEqualsEqualsToken:
+          case ts.SyntaxKind.ExclamationEqualsEqualsToken:
+            if (operatorKind === ts.SyntaxKind.ExclamationEqualsEqualsToken) this.emit('!');
+            this.emit('identical (');
+            this.visit(binExpr.left);
+            this.emit(',');
+            this.visit(binExpr.right);
+            this.emit(')');
+            break;
+          case ts.SyntaxKind.CaretToken:
+          case ts.SyntaxKind.BarToken:
+          case ts.SyntaxKind.AmpersandToken:
+          case ts.SyntaxKind.GreaterThanGreaterThanToken:
+          case ts.SyntaxKind.LessThanLessThanToken:
+          case ts.SyntaxKind.CaretEqualsToken:
+          case ts.SyntaxKind.BarEqualsToken:
+          case ts.SyntaxKind.AmpersandEqualsToken:
+          case ts.SyntaxKind.GreaterThanGreaterThanEqualsToken:
+          case ts.SyntaxKind.LessThanLessThanEqualsToken:
+            // we special case this for Dart since dart2ts will
```

Comment:
dart2ts?

Maybe formulate as `the bitwise operators are only defined on int, so the number types ts2dart deals with have to be converted to int explicitly to match JavaScript's semantics in Dart`?